clients/common/lapack/testing_gesvd.hpp

```diff
@@ -206,23 +206,38 @@ void gesvd_initData(const rocblas_handle handle,
                     const rocblas_int bc,
                     Th& hA,
                     std::vector<T>& A,
-                    bool test = true)
+                    const bool test,
+                    const bool singular)
 {
     if(CPU)
     {
         rocblas_init<T>(hA, true);
 
         for(rocblas_int b = 0; b < bc; ++b)
         {
-            // scale A to avoid singularities
-            for(rocblas_int i = 0; i < m; i++)
+            if(!singular)
             {
-                for(rocblas_int j = 0; j < n; j++)
+                // scale A to avoid singularities
+                for(rocblas_int i = 0; i < m; i++)
                 {
-                    if(i == j)
-                        hA[b][i + j * lda] += 400;
-                    else
-                        hA[b][i + j * lda] -= 4;
+                    for(rocblas_int j = 0; j < n; j++)
+                    {
+                        if(i == j)
+                            hA[b][i + j * lda] += 400;
+                        else
+                            hA[b][i + j * lda] -= 4;
+                    }
+                }
+            }
+            else
+            {
+                // form a singular matrix consisting of all ones
```

Comment:
Setting the matrix to all ones is equivalent to forming the rank-1 matrix 
A = e_m * e_n',  
where e_m is m by 1 column vector with all ones.   

Perhaps a  more general way might be to form  a rank-k matrix as 
A = Uk * Dk * Vk',  where Dk is k by k diagonal matrix with positive diagonal values in descending order, Uk is m by k and Vk is n by k, k <= min(m,n). 

Furthermore if Uk and Vk has orthonormal columns, then this is similar to the SVD of A.     Just a minor suggestion.